pol-core/pol/uoscrobj.cpp

```diff
@@ -1595,6 +1600,37 @@ BObjectImp* Item::set_script_member_id_double( const int id, double value )
   case MBR_QUALITY:
     quality( value );
     return new Double( value );
+  case MBR_WEIGHT_MULTIPLIER_MOD:
+  {
+    if ( value < 0.0 )
+    {
+      return new BError( "weight_multiplier_mod cannot be less than 0.0" );
+    }
+
+    if ( container )
+    {
+      int current_weight = weight();
+      int will_be_new_weight = weight_of( amount_, value );
```

Comment:
That seems to be the only spot where the new weight_of method is called. Since you store the original weight before why not directly set the new weight? This would remove the need of the new function.

Reply:
Addressed in [288f7f4](https://github.com/polserver/polserver/pull/633/commits/288f7f4e923618eb3551843b0bb804644c5c154a).